test/functorch/test_vmap.py

```diff
@@ -5152,6 +5161,145 @@ def check_gte_0(t):
 
         check_vmap_fallback(self, test, torch._test_check_tensor)
 
+class TestVmapNestedTensor(Namespace.TestVmapBase):
+    def _vmap_test(self, *args, **kwargs):
+        return _vmap_test(self, *args, **kwargs)
+
+    # dims should be something like [5, None, 10], with None indicating that a
+    # random ragged structure should be used
+    def _create_nt(self, dims, device):
+        sizes = [
+            [d if d is not None else torch.randint(2, 10, size=(1,)).item() for d in dims[1:]]
+            for d in range(dims[0])
+        ]
+        return torch.nested.nested_tensor([
+            torch.randn(*size) for size in sizes
+        ], device=device)
+
+    # Creates an NT matching another NT's number of components and
+    # shape / ragged structure for all dims specified to be -1.
+    def _nt_from_similar(self, other, dims):
+        assert len(dims) == other.dim()
+        assert dims[0] == -1 or dims[0] == other.size(0)
+
+        ret_sizes = []
+        for t in other.unbind():
+            other_size = t.shape
+            ret_size = []
+            for i, d in enumerate(dims[1:]):
+                if d == -1:
+                    ret_size.append(other_size[i])
+                else:
+                    ret_size.append(d)
+            ret_sizes.append(ret_size)
+
+        return torch.nested.nested_tensor([
+            torch.randn(*size) for size in ret_sizes
+        ], device=other.device)
+
+    @allowVmapFallbackUsage
+    def test_fallback_unary(self, device):
+        def f(x):
+            return x.sin() * 5. + 4.
+
+        nt = self._create_nt([4, None, 3], device=device)
+        self._vmap_test(f, (nt,))
+
```

Comment:
could we add a test for if the Tensor inside of vmap has `is_nested=False`? (I would expect it to have is_nested=False and to look like a regular Tensor, but lmk if you disagree)